converter/src/main/java/gov/cms/qpp/conversion/decode/QrdaDecoderEngine.java

```diff
@@ -170,34 +170,38 @@ private DecodeData decodeChildren(final Element element, final Node parentNode)
 	}
 
 	/**
-	 * Reduces the {@code templateId} {@link Element}s so there are no duplicates.  All other {@link Element}s are left alone.
+	 * Reduces the {@code templateId} {@link Element}s so there are no duplicates. All other {@link Element}s are left alone.
 	 *
 	 * @param childElements The elements to filter
-	 * @return A {@link List} of {@link Element}s that are filtered.
+	 * @return A {@link List} of {@link Element}s that are filtered. If no template ids or at least one implemented template id
+	 * appears within the filtered child elements then the filtered children will be returned otherwise an empty list will
+	 * be returned.
 	 */
 	private List<Element> getUniqueTemplateIdElements(final List<Element> childElements) {
 		Set<TemplateId> uniqueTemplates = EnumSet.noneOf(TemplateId.class);
 
-		return childElements.stream()
+		List<Element> children = childElements.stream()
 			.filter(filterElement -> {
 				boolean isTemplateId = TEMPLATE_ID.equals(filterElement.getName());
 				TemplateId filterTemplateId = getTemplateId(filterElement);
 
 				boolean elementWillStay = true;
 
 				if (isTemplateId) {
-					if (getDecoder(filterTemplateId) == null) {
+					if (getDecoder(filterTemplateId) == null || uniqueTemplates.contains(filterTemplateId)) {
```

Comment:
micro-optimization that you may not like, but you could change the check to uniqueTemplates#add